Source/core/ThreadPool.h

```diff
@@ -85,7 +85,7 @@ namespace Core {
             #pragma warning(default: 4355)
             #endif
 
-            ~JobType()
+            virtual ~JobType()
```

Comment:
Hi,

I fully understand it is used as a base class for Core::IWorkerPool::JobType but I would not expect we do a delete somewhere on the base class (so on a JobType<IMPLEMENTATION> pointer). If we for semantic correctness add the virtual here all the jobs grow in size (as there are no virtuals in JobType yet). You could I guess argue if in this case the public inheritance from JobType is technically the best solution but at this moment we would not prefer to rewrite that.
So was this change created because there really is an issue and somewhere a delete on a JobTYpe<IMPLEMENTATION>* is done or just to make it semantically correct? In the later case we would like to propose to leave it like it is to not grow the size of the JobType objects unnecessarily.

Reply:
Hi @MFransen69! Yes, the main drive for the change was for semantic correctness (no issues detected).

Reply:
Agree to drop this MR